### PR TITLE
Initialize Language Server on Client Connection

### LINKED ISSUE #1403: Language server lazy initialization
### Summary
In our cloud offering, we intend to implement a pool of unassigned language servers. A constructed server will not have any volume attached to a container. Therefore, the Language Server cannot read/write any file during the bootstrap process. The Cloud backend will be leasing a language server on the user's behalf from the pool when a user decides to open a project. Then, the pool controller will attach a durable volume to the container. 

### Value
The cloud offering will be able to spin up runtimes in seconds. 

### Specification
- the language server must not read/write any files on a disk until the `initProtocolConnection` message is received
- upon receiving the `initProtocolConnection` message, the language server may initialize database tables

### Acceptance Criteria & Test Cases
The lazy init logic should be published to the corporate Azure Container Registry.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #1403

Lazy initialization of the Language Server.

Changelog:
- add: `InitializationComponent` containing the init logic
- update: `MainModule` does not create the directories or initializes the databases, or the truffle context
- update: `JsonConnectionController` runs initialization logic when receiving the `InitProtocolConnection` message

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by 4e6** [comment]
Had to remove the failing test because it assumes that the language server is initialized on project/open command. Not proud of that, but I wasn't able to come up with the idea of how to make it work :neutral_face:

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/SequentialResourcesInitialization.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class SequentialResourcesInitialization(
+  resources: Seq[InitializationComponent]
+)(implicit ec: ExecutionContext)
+    extends InitializationComponent {
+
+  /** @inheritdoc */
+  override def init(): Future[InitializationComponent.Initialized.type] =
+    resources.foldLeft(Future.successful(InitializationComponent.Initialized)) {
+      (action, resource) => action.flatMap(_ => resource.init())
+    }
```
Comment (lolczak) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/SequentialResourcesInitialization.scala:17]: `Future` has a combinator for that `Future.sequence`

**Review by lolczak — APPROVED** [review]
Great! Please use `Future.sequence` instead of fold. Can you roll out a new release candidate?

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/SequentialResourcesInitialization.scala (hunk 1 above)
Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/SequentialResourcesInitialization.scala:17]: `Future.sequence` takes `Seq[Future]`, meaning that all futures are started at once. I need sequential behavior, when the next action starts after the previous is complete.

**Review by iamrecursion — APPROVED** [review]
I'm only approving the changes to `.github/settings.yml`.

**Review by radeusgd — CHANGES_REQUESTED** [review]
I've only reviewed the CI config.

It looks good overall, but please update also the `legal-review.yml` to be consistent.

Also `release-update-broken.yml` must definitely be updated because it also uses aws cli.

**Review by radeusgd — APPROVED** [review]
CI config looks good

**Pull request merged**